Source/Core/VideoCommon/TextureConversionShader.cpp

```diff
@@ -720,4 +724,526 @@ const char* GenerateEncodingShader(u32 format, APIType ApiType)
+      R"(
+      layout(local_size_x = 8, local_size_y = 8) in;
+
+      void main()
+      {
+        uvec2 coords = gl_GlobalInvocationID.xy;
+
+        // Tiled in 4x4 blocks
+        // We can't use the normal calculation function, as these are packed as the AR channels
+        // for the entire block, then the GB channels afterwards.
+        uint2 block = coords.xy / 4u;
+        uint2 offset = coords.xy % 4u;
+        uint buffer_pos = u_src_offset;
+
+        // Our buffer has 16-bit elements, so the offsets here are half what they would be in bytes.
+        buffer_pos += block.y * u_src_row_stride;
+        buffer_pos += block.x * 32u;
+        buffer_pos += offset.y * 4u;
+        buffer_pos += offset.x;
+
+        // The two GB channels follow after the block's AR channels.
+        uint val1 = texelFetch(s_input_buffer, int(buffer_pos + 0u)).x;
+        uint val2 = texelFetch(s_input_buffer, int(buffer_pos + 16u)).x;
+
+        uvec4 color;
+        color.a = (val1 & 0xFFu);
+        color.r = (val1 >> 8);
+        color.g = (val2 & 0xFFu);
+        color.b = (val2 >> 8);
+
+        vec4 norm_color = vec4(color) / 255.0;
+        imageStore(output_image, ivec3(ivec2(coords), 0), norm_color);
+      }
+      )"},
+     {GX_TF_CMPR, BUFFER_FORMAT_R32G32_UINT, 0, 64, 1, true,
+      R"(
+      // In the compute version of this decoder, we flatten the blocks to a one-dimension array.
+      // Each group is subdivided into 16, and the first thread in each group fetches the DXT data.
+      // All threads then calculate the possible colors for the block and write to the output image.
+
+      #define GROUP_SIZE 64u
+      #define BLOCK_SIZE_X 4u
+      #define BLOCK_SIZE_Y 4u
+      #define BLOCK_SIZE (BLOCK_SIZE_X * BLOCK_SIZE_Y)
+      #define BLOCKS_PER_GROUP (GROUP_SIZE / BLOCK_SIZE)
+
+      layout(local_size_x = GROUP_SIZE, local_size_y = 1) in;
+
+      shared uvec2 shared_temp[BLOCKS_PER_GROUP];
+
+      uint DXTBlend(uint v1, uint v2)
+      {
+        // 3/8 blend, which is close to 1/3
+        return ((v1 * 3u + v2 * 5u) >> 3);
+      }
+
+      void main()
+      {
+        uint local_thread_id = gl_LocalInvocationID.x;
+        uint block_in_group = local_thread_id / BLOCK_SIZE;
+        uint thread_in_block = local_thread_id % BLOCK_SIZE;
+        uint block_index = gl_WorkGroupID.x * BLOCKS_PER_GROUP + block_in_group;
+
+        // Annoyingly, we can't precalculate this as a uniform because the DXT block size differs
+        // from the block size of the overall texture (4 vs 8). We can however use a multiply and
+        // subtraction to avoid the modulo for calculating the block's X coordinate.
+        uint blocks_wide = u_src_size.x / BLOCK_SIZE_X;
+        uvec2 block_coords;
+        block_coords.y = block_index / blocks_wide;
+        block_coords.x = block_index - (block_coords.y * blocks_wide);
+
+        // Only the first thread for each block reads from the texel buffer.
+        if (thread_in_block == 0u)
```

Comment:
No clue if you care about performance, but this will make all wraps to execute this block. It would be better to check for if (local_thread_id.x < BLOCK_SIZE) to move all execution to the beginning of the threads.